zetaclient/orchestrator/orchestrator.go

```diff
@@ -318,6 +327,13 @@ func (oc *Orchestrator) runScheduler(ctx context.Context) error {
 				continue
 			}
 
+			sleepDuration := time.Until(newBlock.Block.Time.Add(oc.signerBlockTimeOffset))
```

Comment:
Can we abstract this from orchestrator and move it to zetacore client?  E.g.

```
oc.zetacoreClient.NewBlockSubscriber(ctx, true) // syncBlock=true
```

Reply:
Maybe but I'm not sure if we'll want the same latency offset for different processes.

Reply:
> if we'll want the same latency offset for different processes.

In this case, you can add the third param `jitter time.Duration` that would simulate some randomness. 

But the point is that the orchestrator should care about observers/signers/ and this logic is a bit out of its responsibilities IMO